packages/block-editor/src/components/list-view/index.js

```diff
@@ -174,6 +174,11 @@ function ListView(
 		[ isMounted.current, draggedClientIds, expandedState, expand, collapse ]
 	);
 
+	// If there are no blocks to show, do not render the list view.
```

Comment:
@talldan Is there a reason this was not added previously? I found it really confusing to render an empty list view.